src/PSR7/ServerRequestValidator.php

```diff
@@ -4,50 +4,41 @@
 
 namespace OpenAPIValidation\PSR7;
 
-use cebe\openapi\exceptions\TypeErrorException;
-use cebe\openapi\spec\Header as HeaderSpec;
 use cebe\openapi\spec\OpenApi;
+use OpenAPIValidation\PSR7\Exception\MultipleOperationsMismatchForRequest;
 use OpenAPIValidation\PSR7\Exception\NoOperation;
-use OpenAPIValidation\PSR7\Exception\Request\MissedRequestCookie;
-use OpenAPIValidation\PSR7\Exception\Request\MissedRequestHeader;
-use OpenAPIValidation\PSR7\Exception\Request\MissedRequestQueryArgument;
-use OpenAPIValidation\PSR7\Exception\Request\MultipleOperationsMismatchForRequest;
-use OpenAPIValidation\PSR7\Exception\Request\RequestBodyMismatch;
-use OpenAPIValidation\PSR7\Exception\Request\RequestCookiesMismatch;
-use OpenAPIValidation\PSR7\Exception\Request\RequestHeadersMismatch;
-use OpenAPIValidation\PSR7\Exception\Request\RequestPathParameterMismatch;
-use OpenAPIValidation\PSR7\Exception\Request\RequestQueryArgumentMismatch;
-use OpenAPIValidation\PSR7\Exception\Request\Security\MalformattedRequestAuthorizationHeader;
-use OpenAPIValidation\PSR7\Exception\Request\Security\NoRequestAuthorizationHeader;
-use OpenAPIValidation\PSR7\Exception\Request\Security\NoRequestSecurityApiKey;
-use OpenAPIValidation\PSR7\Exception\Request\Security\RequestSecurityMismatch;
-use OpenAPIValidation\PSR7\Exception\Request\UnexpectedRequestContentType;
-use OpenAPIValidation\PSR7\Exception\Request\UnexpectedRequestHeader;
-use OpenAPIValidation\PSR7\Validators\Body;
-use OpenAPIValidation\PSR7\Validators\Cookies;
-use OpenAPIValidation\PSR7\Validators\Headers;
-use OpenAPIValidation\PSR7\Validators\Path;
-use OpenAPIValidation\PSR7\Validators\QueryArguments;
-use OpenAPIValidation\PSR7\Validators\Security;
+use OpenAPIValidation\PSR7\Exception\ValidationFailed;
+use OpenAPIValidation\PSR7\Validators\BodyValidator;
+use OpenAPIValidation\PSR7\Validators\CookiesValidator;
+use OpenAPIValidation\PSR7\Validators\HeadersValidator;
+use OpenAPIValidation\PSR7\Validators\PathValidator;
+use OpenAPIValidation\PSR7\Validators\QueryArgumentsValidator;
+use OpenAPIValidation\PSR7\Validators\SecurityValidator;
+use OpenAPIValidation\PSR7\Validators\ValidatorChain;
 use Psr\Http\Message\ServerRequestInterface;
 use Throwable;
 use function count;
-use function json_decode;
-use function json_encode;
-use function property_exists;
 use function strtolower;
 
 class ServerRequestValidator implements ReusableSchema
 {
     /** @var OpenApi */
     protected $openApi;
-    /** @var SpecFinder */
-    protected $finder;
+    /** @var MessageValidator */
+    protected $validator;
 
     public function __construct(OpenApi $schema)
     {
-        $this->openApi = $schema;
-        $this->finder  = new SpecFinder($this->openApi);
+        $this->openApi   = $schema;
+        $finder          = new SpecFinder($this->openApi);
+        $this->validator = new ValidatorChain(
```

Comment:
Most important changes here. I've replace class methods with validator chain with separate validators. I have some thoughts about configuring the chain outside of the class, but not in this PR